Update to rails 4 2

davidbasalla opened the pull request:
For https://trello.com/c/yd6NOD9O/156-upgrade-rails-on-licence-finder

Upgrade to the app to Rails 4.2, along with Mongoid and RSpec.

dsingleton commented:
I've not looked at the changes to the app itself yet, but this is going to need a deployment configuration change for

 - Secret key, `secret_token` / `secrets.yml`
 - Mongo config for `mongoid.yml`

We have some prior art for these changes on https://github.com/alphagov/asset-manager/pull/42, so once we've made/tested the changes there we can do the same for this app.

I'll try and look at the actual app changes this afternoon.

dsingleton commented on config/secrets.yml:
```diff
@@ -0,0 +1,16 @@
+development:
+  secret_key_base: '54dec6e791dce377c9b3a28aca520b427be2b469d3f4f4632158435a7edf0e3267fadaf94520d5dd5910116a227d6c7904bc81164de4f97a6549aea144dec46d'
+test:
+  secret_key_base: '54dec6e791dce377c9b3a28aca520b427be2b469d3f4f4632158435a7edf0e3267fadaf94520d5dd5910116a227d6c7904bc81164de4f97a6549aea144dec46d'
```
This should have a line for production, reading from ENV, (heres an [example from asset-manager](https://github.com/alphagov/asset-manager/commit/6f7c5971c7e6b080c40e1aa573be047fc31cf2cf)

It'll need a change in `puppet`/`deployment`, which i'm looking at, to populate the ENV.

dsingleton commented:
It looks like the asset precompile step is failing with `"NameError: uninitialized constant Uglifier::VERSION"]`, which seems to be caused by using an older version of `uglify`.

Some frontend apps [include compiling assets](https://github.com/alphagov/frontend/blob/master/jenkins.sh#L29) in the jenkins test runner, to catch this earlier. I think that makes sense for most of our frontend apps, and is worth adding here too (assuming it doesn't massively slow jenkins builds)

davidbasalla commented:
@dsingleton I have updated secrets.yml to add the production env var placeholder. 
I've also updated the uglifier gem and added the precompile step to jenkins.sh. Do you mind having a look at https://github.com/alphagov/licence-finder/commit/6029e4e33ca4c03cfe8cb73b978852c22d2c6bae? I've not worked much with setting up jenkins so not sure if this is correct.

dsingleton commented:
> I've also updated the uglifier gem and added the precompile step to jenkins.sh. Do you mind having a look at 6029e4e? I've not worked much with setting up jenkins so not sure if this is correct.

Looks correct to me, and CI passes :+1: 

Everything else LGTM, assuming it's working on preview.

dsingleton commented:
The deployment config change has been merged, so happily merging.
